tsl/src/continuous_aggs/create.c

```diff
@@ -1751,44 +1751,59 @@ cagg_create(ViewStmt *stmt, Query *panquery, CAggTimebucketInfo *origquery_ht,
 	return;
 }
 
-/* entry point for creating continuous aggregate view
- * step 1 : validate query
- * step 2: create underlying tables and views
- */
-DDLResult
-tsl_process_continuous_agg_viewstmt(ViewStmt *stmt, const char *query_string, void *pstmt,
-									WithClauseResult *with_clause_options)
+static bool
+continuous_agg_process_create_stmt(CreateTableAsStmt *stmt, const char *query_string, void *pstmt,
+								   WithClauseResult *with_clause_options)
 {
-	Query *query = NULL;
 	CAggTimebucketInfo timebucket_exprinfo;
 	Oid nspid;
-	PlannedStmt *pstmt_info = (PlannedStmt *) pstmt;
-	RawStmt *rawstmt = NULL;
-	/* we have a continuous aggregate query. convert to Query structure
-	 */
-	rawstmt = makeNode(RawStmt);
-	rawstmt->stmt = (Node *) copyObject(stmt->query);
-	rawstmt->stmt_location = pstmt_info->stmt_location;
-	rawstmt->stmt_len = pstmt_info->stmt_len;
-	query = parse_analyze(rawstmt, query_string, NULL, 0, NULL);
+	ViewStmt viewstmt = {
+		.type = T_ViewStmt,
+		.view = stmt->into->rel,
+		.query = stmt->into->viewQuery,
+		.options = stmt->into->options,
+		.aliases = stmt->into->colNames,
+	};
 
-	nspid = RangeVarGetCreationNamespace(stmt->view);
-	if (get_relname_relid(stmt->view->relname, nspid))
+	nspid = RangeVarGetCreationNamespace(stmt->into->rel);
+
+	if (get_relname_relid(stmt->into->rel->relname, nspid))
```

Comment:
Shouldn't this honor `IF NOT EXISTS`? (Looks like we are always failing here in case of an existing aggregate). Do we have a test for it?

Reply:
Added check for it working in a similar way to how it works for normal tables. And added test case.